### PR TITLE
Remove outdated info about old releases 2.6 and 1.6

### PR DESCRIPTION
**Pull request by macedogm**
## Issue: 

This PR updates/removes outdated info about releases 2.6 and 1.6:

1. Update `2.6` to `2.8` in `README.md` and `README-template.md`.
2. Remove `1.6` from the template `README-template.md`.
3. Remove `README_1_6.md` entirely.

 
### Automated Testing

* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

**Review by deniseschannon — DISMISSED** [review]

**Review by cbron — DISMISSED** [review]

File: .github/workflows/update-readme.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
        tmpfile=$(mktemp)
        unset latest
        unset stable
-       for rancher_major_minor in 2.7 2.6 2.5; do
+       for rancher_major_minor in 2.7 2.6; do
```
Comment (cbron) [inline .github/workflows/update-readme.yml:23]: Looks like this was done already: https://github.com/rancher/rancher/blob/release/v2.8/.github/workflows/update-readme.yml#L23

File: README.md [hunk 2]
```diff
@@ -17,9 +17,6 @@ Rancher is an open source container management platform built for organizations
 * v2.6
   * Latest - v2.6.13 - `rancher/rancher:v2.6.13` - Read the full release [notes](https://github.com/rancher/rancher/releases/tag/v2.6.13).
   * Stable - v2.6.13 - `rancher/rancher:v2.6.13` - Read the full release [notes](https://github.com/rancher/rancher/releases/tag/v2.6.13).
-* v2.5
-  * Latest - v2.5.17 - `rancher/rancher:v2.5.17` - Read the full release [notes](https://github.com/rancher/rancher/releases/tag/v2.5.17).
-  * Stable - v2.5.17 - `rancher/rancher:v2.5.17` - Read the full release [notes](https://github.com/rancher/rancher/releases/tag/v2.5.17).
```
Comment (cbron) [inline README.md:22]: and this was done

**Review by cbron — COMMENTED** [review]
Only thing remaining in this PR is removing 1.6 info, but that's not a big deal either way.

**Review by cbron — APPROVED** [review]

**Comment by macedogm** [comment]
@cbron unsure what to do here, as the pipeline is constantly failing due to flaky tests (see https://github.com/rancher/rancher/issues/43560) and which are no needed for these markdown related changes.

**Comment by cbron** [comment]
I'll just hard merge, since this only affects markdown files, and I'd like it to be up to date for 2.8.0.

**Pull request merged**